tests/python_package_test/test_engine.py

```diff
@@ -48,2591 +49,2655 @@ def categorize(continuous_x):
+
+
+def test_categorical_non_zero_inputs():
+    x = [1, 1, 1, 1, 1, 1, 2, 2]
+    y = [1, 1, 1, 1, 1, 1, 0, 0]
+
+    X_train = np.array(x).reshape(len(x), 1)
+    y_train = np.array(y)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    lgb_eval = lgb.Dataset(X_train, y_train)
+
+    params = {
+        'objective': 'regression',
+        'metric': 'auc',
+        'verbose': -1,
+        'boost_from_average': False,
+        'min_data': 1,
+        'num_leaves': 2,
+        'learning_rate': 1,
+        'min_data_in_bin': 1,
+        'min_data_per_group': 1,
+        'cat_smooth': 1,
+        'cat_l2': 0,
+        'max_cat_to_onehot': 1,
+        'zero_as_missing': False,
+        'categorical_column': 0
+    }
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=1,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    pred = gbm.predict(X_train)
+    np.testing.assert_allclose(pred, y)
+    ret = roc_auc_score(y_train, pred)
+    assert ret > 0.999
+    assert evals_result['valid_0']['auc'][-1] == pytest.approx(ret)
+
+
+def test_multiclass():
+    X, y = load_digits(n_class=10, return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
+        'num_class': 10,
+        'verbose': -1
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, params=params)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train, params=params)
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=50,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    ret = multi_logloss(y_test, gbm.predict(X_test))
+    assert ret < 6
+    assert evals_result['valid_0']['multi_logloss'][-1] == pytest.approx(ret)
+
+
+def test_multiclass_rf():
+    X, y = load_digits(n_class=10, return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'boosting_type': 'rf',
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
+        'bagging_freq': 1,
+        'bagging_fraction': 0.6,
+        'feature_fraction': 0.6,
+        'num_class': 10,
+        'num_leaves': 50,
+        'min_data': 1,
+        'verbose': -1,
+        'gpu_use_dp': True
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, params=params)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train, params=params)
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=50,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    ret = multi_logloss(y_test, gbm.predict(X_test))
+    assert ret < 3
```

Comment:
```suggestion
    assert ret < 0.23
```

from https://www.textcompare.org/index.html?id=6008e19c6d59840017ac7a56

![image](https://user-images.githubusercontent.com/7608904/105270573-9b43de80-5b5b-11eb-8c4d-5b6d14ad730a.png)